include/wasmtime.hh

```diff
@@ -2454,12 +2472,14 @@ public:
    * > signature is statically known it's recommended to use `Func::typed` and
    * > `TypedFunc::call`.
    */
+  template<typename I>
   TrapResult<std::vector<Val>> call(Store::Context cx,
-                                    const std::vector<Val> &params) const {
+                                    const I &begin,
+                                    const I &end) const {
     std::vector<wasmtime_val_t> raw_params;
-    raw_params.reserve(params.size());
-    for (const auto &param : params) {
-      raw_params.push_back(param.val);
+    raw_params.reserve(end - begin);
```

Comment:
You might want to static assert or conditionally do this only if it's a random access iterator.

Reply:
Oh good point, what's the best way to do that in C++?